homeassistant/components/iotty/switch.py

```diff
@@ -20,31 +32,113 @@
 
 _LOGGER = logging.getLogger(__name__)
 
+T = TypeVar("T", bound=Device)
+
+
+@dataclass(frozen=True, kw_only=True)
+class IottySwitchEntityDescription(
+    Generic[T],
+    SwitchEntityDescription,
+):
+    """Description of a iotty Switch."""
+
+    control_fn: Callable[[IottyProxy, T, str], Coroutine[Any, Any, Any]]
+    is_on_fn: Callable[[T], bool]
```

Comment:
Hi, this is actually needed because the base `Device` type doesn't have the `is_on` property, so I cannot access it if I declare `T` as `Device`. I could remove the generic from `control_fn`, but it wouldn't be consistent and also not future-proof, so I'd rather have them both as generics.

Reply:
I agree the way generic is used here seems unnecessary

Reply:
Thanks for the input, I have removed it